### ISSUE TITLE
NullReferenceException at OpenRA.Chat.GlobalChat.OnJoin

### ISSUE THREAD

**Issue by r34ch** [OpenRA/OpenRA#10338]
Can't remember what hash, but I very quickly clicked on single player then skirmish (i believe)

```
Red Alert Mod at Version {DEV_VERSION}
Operating System: Linux (Unix 3.13.0.24)
Runtime Version: Mono 4.2.1 (Stable 4.2.1.102/6dd2d0d Thu Nov 12 09:52:44 UTC 2015) CLR 4.0.30319.17020
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object
  at OpenRA.Chat.GlobalChat.OnJoin (System.Object sender, Meebey.SmartIrc4net.JoinEventArgs e) <0x40364500 + 0x000ef> in <filename unknown>:0 
  at Meebey.SmartIrc4net.IrcClient._Event_JOIN (Meebey.SmartIrc4net.IrcMessageData ircdata) <0x403635e0 + 0x0028f> in <filename unknown>:0 
  at Meebey.SmartIrc4net.IrcClient._HandleEvents (Meebey.SmartIrc4net.IrcMessageData ircdata) <0x4034c470 + 0x0041b> in <filename unknown>:0 
  at Meebey.SmartIrc4net.IrcClient._Worker (System.Object sender, Meebey.SmartIrc4net.ReadLineEventArgs e) <0x4034a2d0 + 0x0002f> in <filename unknown>:0 
  at (wrapper delegate-invoke) <Module>:invoke_void_object_ReadLineEventArgs (object,Meebey.SmartIrc4net.ReadLineEventArgs)
  at Meebey.SmartIrc4net.IrcConnection.ReadLine (Boolean blocking) <0x40345560 + 0x00175> in <filename unknown>:0 
  at Meebey.SmartIrc4net.IrcConnection.Listen (Boolean blocking) <0x403454f0 + 0x0002f> in <filename unknown>:0 
  at Meebey.SmartIrc4net.IrcConnection.Listen () <0x403454c0 + 0x00013> in <filename unknown>:0 
  at OpenRA.Chat.GlobalChat.<Connect>m__8 () <0x40331060 + 0x001d3> in <filename unknown>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) <0x7f2d748b1010 + 0x00099> in <filename unknown>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, Boolean preserveSyncCtx) <0x7f2d748af850 + 0x0016e> in <filename unknown>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, Boolean preserveSyncCtx) <0x7f2d748af820 + 0x00020> in <filename unknown>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) <0x7f2d748af770 + 0x00059> in <filename unknown>:0 
  at System.Threading.ThreadHelper.ThreadStart () <0x7f2d748b11d0 + 0x0002e> in <filename unknown>:0
```

**Comment by RoosterDragon** [comment]
Perhaps we should consider adding a try-catch canary to this so we can work out what is null when it next occurs.

**Comment by abcdefg30** [comment]
Should be fixed by #11238.
